src/main/java/seedu/address/logic/commands/AddAppointmentCommand.java

```diff
@@ -0,0 +1,121 @@
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.appointment.Appointment;
+import seedu.address.model.appointment.Time;
+import seedu.address.model.doctor.Doctor;
+import seedu.address.model.patient.Patient;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.shared.Date;
+
+/**
+ * Adds an appointment to an existing person in the address book.
+ */
+public class AddAppointmentCommand extends Command {
+
+    public static final String COMMAND_WORD = "add-appt";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Adds an appointment to the patient and doctor identified "
+            + "at the specified date and time.\n"
+            + "Parameters: "
+            + PREFIX_PATIENT_NAME + "PATIENT NAME "
+            + PREFIX_DOCTOR_NAME + "DOCTOR NAME "
+            + PREFIX_DATE + "DATE "
+            + PREFIX_TIME + "TIME\n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_PATIENT_NAME + "John Doe "
+            + PREFIX_DOCTOR_NAME + "Jane Doe "
+            + PREFIX_DATE + "23-09-2024 "
+            + PREFIX_TIME + "1100";
+
+    public static final String MESSAGE_SUCCESS = "New appointment added: %1$s";
+    public static final String MESSAGE_DUPLICATE_APPOINTMENT = "This appointment already exists in the address book";
+
+    private final Name patientName;
+    private final Name doctorName;
+    private final Date date;
+    private final Time time;
+    private Appointment appointmentToAdd;
+
+    /**
+     * Creates an AddAppointmentCommand to add the specified {@code Appointment}
+     */
+    public AddAppointmentCommand(Name patientName, Name doctorName, Date date, Time time) {
+        requireAllNonNull(patientName, doctorName, date, time);
+        this.patientName = patientName;
+        this.doctorName = doctorName;
+        this.date = date;
+        this.time = time;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        Optional<Person> patient = lastShownList.stream()
+                .filter(person -> person.getName().equals(patientName))
+                .findAny();
+        Optional<Person> doctor = lastShownList.stream()
+                .filter(person -> person.getName().equals(doctorName))
+                .findAny();
+
+        Person patientToEdit = patient.get();
+        Person doctorToEdit = doctor.get();
+        this.appointmentToAdd = new Appointment((Doctor) doctorToEdit, (Patient) patientToEdit, this.date, this.time);
+
+        Person editedPatient;
+        Person editedDoctor;
+
+        if (patientToEdit.addAppointment(appointmentToAdd) && doctorToEdit.addAppointment(appointmentToAdd)) {
```

Comment:
I believe in Brandon's implementation, the constructor of `Appointment` already calls `patient.addAppointment` and `doctor.addAppointment`. Since appointment is already made, both methods are already called before these 2 are done, causing it to always return false?

![image](https://github.com/user-attachments/assets/d96b8ad0-e511-4941-8079-4514326a540c)